Zend/zend_vm_def.h

```diff
@@ -5885,9 +6039,15 @@ ZEND_VM_HANDLER(125, ZEND_FE_RESET_RW, CONST|TMP|VAR|CV, JMP_ADDR)
 			ZEND_VM_NEXT_OPCODE_CHECK_EXCEPTION();
 		} else {
 			zend_class_entry *ce = Z_OBJCE_P(array_ptr);
-			zend_object_iterator *iter = ce->get_iterator(ce, array_ptr, 1);
+			zend_object_iterator *iter;
 			zend_bool is_empty;
 
+			if (UNEXPECTED(ZEND_CLASS_HAS_TYPE_HINTS(ce))) {
+				ZEND_VM_C_GOTO(fe_reset_rw_typed);
+			}
```

Comment:
Why this? `get_iterator` is not iterating over the properties, so it should be fine.